core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -0,0 +1,77 @@
+package org.infinispan;
+
+import java.util.Collection;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+
+import org.infinispan.configuration.cache.TransactionConfiguration;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.util.function.SerializableBiConsumer;
+import org.infinispan.util.function.SerializablePredicate;
+
+/**
+ * Stream that allows for operation upon data solely with side effects by using {@link LockedStream#forEach(BiConsumer)}
+ * where the <b>BiConsumer</b> is invoked while guaranteeing that the entry being passed is properly locked for the
+ * entire duration of the invocation.
```

Comment:
I personally hate that usage. It is really tailored for single node modifications. I can double check by adding a test if this works properly, but it should though.